### ISSUE TITLE
Recovering APPLICATION_METHOD dynatrace_entities broken in upgrading from 1.44.0 to 1.44.1

### ISSUE THREAD

**Issue by mattBaumBeneva** [dynatrace-oss/terraform-provider-dynatrace#343]
**Describe the bug**
We sequentially create the following entities in our Web Application configuration:
Web Application (with Key use actions) -> Apdex (for some KUA) -> SLO (for KUA with Apdex) -> Site Reliability Guardian (based on SLO metrics)

In provider version 1.44.0, this chain of creations can be completed. Though 1.44.1 corrects some bugs, we notice that querying our KUAs like so no longer functions:

```
data "dynatrace_entities" "kuas" {
  entity_selector = "type(\"APPLICATION_METHOD\"),fromRelationships.isApplicationMethodOf(entityId(\"${dynatrace_web_application.application.id}\"))"
}
```

Later in our module, we use the value returned by this like:
`for kua in coalesce(data.dynatrace_entities.kuas.entities, [])`

We output coalesce(data.dynatrace_entities.kuas.entities, []) for testing.
In 1.44.0 we get:

```
15:12:53  foo = tolist([
15:12:53    {
15:12:53      "display_name" = "loading of page /test/transaction2"
15:12:53      "entity_id" = "APPLICATION_METHOD-541C64F3012B0A6D"
15:12:53      "properties" = tomap({
15:12:53        "detectedName" = "loading of page /test/transaction2"
15:12:53      })
15:12:53      "tags" = tolist([])
15:12:53      "type" = "APPLICATION_METHOD"
15:12:53    },
15:12:53    {
15:12:53      "display_name" = "loading of page /test/transaction1"
15:12:53      "entity_id" = "APPLICATION_METHOD-F87A33615F9492DF"
15:12:53      "properties" = tomap({
15:12:53        "detectedName" = "loading of page /test/transaction1"
15:12:53      })
15:12:53      "tags" = tolist([])
15:12:53      "type" = "APPLICATION_METHOD"
15:12:53    },
15:12:53  ])
```

In 1.44.1, we get:

`15:25:58  foo = tolist([])`

**Expected behavior**
We must still be able to apply each of these entities in sequence, even if none of the configurations exist beforehand.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
One additional side note here. If you're using this syntax (essentially not referring to the `entities` directly)
```
output "foo" {
    value = data.dynatrace_entities.kuas
}
```
the results will work already with the current version. In that case Terraform isn't allowed to assume that the value will be final *before* invoking the data source. I'm just not sure whether you'll be able to use that within your module.

**Comment by mattBaumBeneva** [comment]
I think your fix will correct things for us. We are using the entities collection within for and for_each statements.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hello @mattBaumBeneva, in addition to that v1.45.0 also introduced the resource `dynatrace_key_user_action`.
In other words, you don't necessarily have to rely on the data source `dynatrace_entities` for the IDs of the key user actions.

**Comment by mattBaumBeneva** [comment]
@Dynatrace-Reinhard-Pilz , I tried applying with the new resource, but I obtained an error:

```
13:16:01  │ Error: Provider produced inconsistent result after apply
13:16:01  │ 
13:16:01  │ When applying changes to
13:16:01  │ module.Application_test-de-base.dynatrace_key_user_action.kua_without_apdex["loading
13:16:01  │ of page /test/transaction1"], provider
13:16:01  │ "provider[\"registry.terraform.io/dynatrace-oss/dynatrace\"]" produced an
13:16:01  │ unexpected new value: Root resource was present, but now absent.
13:16:01  │ 
13:16:01  │ This is a bug in the provider, which should be reported in the provider's
13:16:01  │ own issue tracker.
```

**Comment by mattBaumBeneva** [comment]
After destroying and applying again, I did not reproduce this message. I'm not certain what the cause was.

**Comment by mattBaumBeneva** [comment]
@Dynatrace-Reinhard-Pilz , I also notice an error on one of our modules, with this code:

```
data "dynatrace_entities" "process_groups" {
  entity_selector = "type(\"PROCESS_GROUP\"),entityName(\"${var.nom}\"),mzId(\"${local.proprietaire.legacy_id}\")"
}

resource "dynatrace_pg_alerting" "alerting" {
  count = length(data.dynatrace_entities.process_groups.entities)

  enabled       = var.alertage.disponibilite.actif
  process_group = data.dynatrace_entities.process_groups.entities[count.index].id

  alerting_mode = "ON_PGI_UNAVAILABILITY"
}
```

The error is:
```
14:22:39  │ Error: Invalid count argument
14:22:39  │ 
14:22:39  │   on ../../src/modules/blueprints/Processus/main.tf line 35, in resource "dynatrace_pg_alerting" "alerting":
14:22:39  │   35:   count = length(data.dynatrace_entities.process_groups.entities)
14:22:39  │ 
14:22:39  │ The "count" value depends on resource attributes that cannot be determined
14:22:39  │ until apply, so Terraform cannot predict how many instances will be
14:22:39  │ created. To work around this, use the -target argument to first apply only
14:22:39  │ the resources that the count depends on.
```

I'm puzzled, because we have modules that follow a similar pattern, with a different entity selector, and are planned and applied correctly.

**Comment by mattBaumBeneva** [comment]
@Dynatrace-Reinhard-Pilz , please ignore the above comments. This issue was fixed by upgrading Terraform from 1.5.4 to 1.6.2
